agent/agent.go

```diff
@@ -525,14 +526,34 @@ func New(options ...AgentOption) (*Agent, error) {
 		return nil, fmt.Errorf("Failed to setup node ID: %v", err)
 	}
 
-	acOpts := []autoconf.Option{
-		autoconf.WithDirectRPC(a.connPool),
-		autoconf.WithTLSConfigurator(a.tlsConfigurator),
-		autoconf.WithBuilderOpts(flat.builderOpts),
-		autoconf.WithLogger(a.logger),
-		autoconf.WithOverrides(flat.overrides...),
+	// We used to do this in the Start method. However it doesn't need to go
+	// there any longer. Originally it did because we passed the agent
+	// delegate to some of the cache registrations. Now we just
+	// pass the agent itself so its safe to move here.
+	a.registerCache()
+
+	cmConf := new(certmon.Config).
+		WithCache(a.cache).
+		WithTLSConfigurator(a.tlsConfigurator).
+		WithDNSSANs(a.config.AutoConfig.DNSSANs).
+		WithIPSANs(a.config.AutoConfig.IPSANs).
+		WithDatacenter(a.config.Datacenter).
+		WithNodeName(a.config.NodeName).
+		WithFallback(a.autoConfigFallbackTLS).
+		WithLogger(a.logger.Named(logging.AutoConfig)).
+		WithTokens(a.tokens)
+	acCertMon, err := certmon.New(cmConf)
```

Comment:
What does having two cert-monitors running at the same time mean? If I have both auto-config and auto-encrypt setup, are the two `CertMonitor` instances overwriting each other's certificates in the TLS Configurator?

Reply:
Apparently I should disable auto-encrypt if auto-config is enabled. They would be overwriting each others certificates.

Reply:
I added some code into the config validation to explicitly fail if yo enable both auto encrypt and auto config.